src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisTest.java

```diff
@@ -3276,4 +3276,27 @@ void testPhaseTapChangerContingency() {
         assertActivePowerEquals(50.0, network.getBranch("L3").getTerminal1());
         assertActivePowerEquals(0.0, network.getBranch("L4").getTerminal1());
     }
+
+    @Test
+    void testMultiThreads() {
+        Network network = createNodeBreakerNetwork();
+        network.getVariantManager().allowVariantMultiThreadAccess(true);
+
+        LoadFlowParameters lfParameters = new LoadFlowParameters();
+        setSlackBusId(lfParameters, "VL1_1");
+        SecurityAnalysisParameters securityAnalysisParameters = new SecurityAnalysisParameters();
+        securityAnalysisParameters.setLoadFlowParameters(lfParameters);
+        OpenSecurityAnalysisParameters securityAnalysisParametersExt = new OpenSecurityAnalysisParameters();
+        securityAnalysisParameters.addExtension(OpenSecurityAnalysisParameters.class, securityAnalysisParametersExt);
+
+        List<Contingency> contingencies = Stream.of("L1", "L2")
+                .map(id -> new Contingency(id, new BranchContingency(id)))
+                .toList();
+
+        SecurityAnalysisResult resultOneThread = runSecurityAnalysis(network, contingencies, Collections.emptyList(), securityAnalysisParameters);
+        securityAnalysisParametersExt.setThreadCount(2);
+        SecurityAnalysisResult resultTwoThreads = runSecurityAnalysis(network, contingencies, Collections.emptyList(), securityAnalysisParameters);
```

Comment:
Note that this test won't detect any race condition issue in case some may happen today or in the future.
The number of threads is too small.